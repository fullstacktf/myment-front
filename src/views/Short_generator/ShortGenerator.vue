<template>
  <div class="columns bigParent">
    <div class="column void1">
      <router-link to="/home">
        <img src="../../../public/assets/myment-logo.png" alt="MYMENT" />
      </router-link>
    </div>
    <div class="column centerzone is-10">
      <b-field class="topField">
        <b-taginput v-model="tags" type="is-dark" placeholder="Enter the location in 2 tags "></b-taginput>
        <b-radio-button>search location</b-radio-button>
      </b-field>
      <div class="tabWrapper">
        <b-tabs type="is-toggle is-danger " expanded>
          <b-tab-item label="short" icon="video">
            <div class="masterContent">
              <!-- this is the component of the itinerary -->
              <div class="itineraryComponent" v-if="tagsAdded">
                <div class="ideaTableShort">
                  <idea-item
                    v-for="idea in ideaStructure"
                    :name="idea.name"
                    :key="idea.id"
                    :description="idea.description"
                    :start="idea.timeStart"
                    :end="idea.timeEnd"
                  ></idea-item>
                </div>
              </div>
              <div class="columns" v-else>
                <div class="column is-3"></div>
                <div class="column is-6">
                  <button
                    v-if="!tagsAdded"
                    class="button is-primary is-large is-warning is-fullwidth"
                    @click="isComponentModalActive = true"
                  >Generate Itinerary !</button>

                  <b-modal :active.sync="isComponentModalActive" has-modal-card>
                    <tag-adder></tag-adder>
                  </b-modal>
                </div>
                <div class="column is-3"></div>
              </div>
            </div>
          </b-tab-item>
          <b-tab-item label="custom" icon="library-music">
            <div class="ideaTableCustom">
              <h1>custom itinerary not available</h1>
            </div>
          </b-tab-item>
          <b-tab-item label="free" icon="video">
            <div class="ideaTableFree">
              <h1>free itinerary not available</h1>
            </div>
          </b-tab-item>
        </b-tabs>
      </div>
    </div>
    <div class="column void2">
      <router-link to="/">
        <a class="delete is-medium"></a>
      </router-link>
    </div>
  </div>
</template>

<script>
import Button from '../../components/Button';
import IdeaItem from '../../components/IdeaItem';
import TagAdder from '../Tag_adder/TagAdder';

export default {
  name: 'Short',
  components: {
    'option-button': Button,
    'tag-adder': TagAdder,
    'idea-item': IdeaItem,
  },
  data() {
    return {
      ideaStructure: [
        {
          name: 'hotel',
          description: 'lorem ipsum ...',
          timeStart: 0,
          timeEnd: 0,
        },
        {
          name: 'hotel',
          description: 'lorem ipsum ...',
          timeStart: 0,
          timeEnd: 0,
        },
        {
          name: 'hotel',
          description: 'lorem ipsum ...',
          timeStart: 0,
          timeEnd: 0,
        },
        {
          name: 'hotel',
          description: 'lorem ipsum ...',
          timeStart: 0,
          timeEnd: 0,
        },
        {
          name: 'hotel',
          description: 'lorem ipsum ...',
          timeStart: 0,
          timeEnd: 0,
        },
        {
          name: 'hotel',
          description: 'lorem ipsum ...',
          timeStart: 0,
          timeEnd: 0,
        },

        {
          name: 'hotel',
          description: 'lorem ipsum ...',
          timeStart: 0,
          timeEnd: 0,
        },

        {
          name: 'hotel',
          description: 'lorem ipsum ...',
          timeStart: 0,
          timeEnd: 0,
        },

        {
          name: 'hotel',
          description: 'lorem ipsum ...',
          timeStart: 0,
          timeEnd: 0,
        },

        {
          name: 'hotel',
          description: 'lorem ipsum ...',
          timeStart: 0,
          timeEnd: 0,
        },
      ],
      tagsAdded: false,
      isComponentModalActive: false,
    };
  },
};
</script>

<style scoped>
.bigParent {
}

.topField {
  margin-top: 20px;
}

.void1 {
  padding: 50px;
  display: flex;
  justify-content: center;
  align-items: flex-start;
}

.tabWrapper {
  border-radius: 5px;
  background-color: aliceblue;
}

.ideaTableCustom {
  display: flex;
  flex-wrap: wrap;

  font-size: 100px;
  flex-flow: column;
  justify-content: center;
  align-items: center;
  max-height: 600px;
  overflow: hidden scroll;
  background-color: white;
}
.tagComponent {
  align-self: center;
}

.ideaTableFree {
  display: flex;
  font-size: 100px;
  flex-wrap: wrap;
  flex-flow: column;
  justify-content: center;
  align-items: center;
  max-height: 600px;
  overflow: hidden scroll;
  background-color: white;
}

.ideaTableShort {
  display: flex;
  flex-wrap: wrap;
  flex-flow: column;
  max-height: 600px;
  overflow: hidden scroll;
}

.timeLine {
  max-height: 20%;
  overflow: hidden;
}
</style>
