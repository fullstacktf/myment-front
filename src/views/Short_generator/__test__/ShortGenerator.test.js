import ShortGenerator from '../ShortGenerator.vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import storeConfig from '../../../store/store-index';
import { cloneDeep } from 'lodash';
import Buefy from 'buefy';
import { mock } from '../../../test/mockBack';

describe('ShortGenerator', () => {
  let localVue;
  beforeEach(() => {
    const Vue = createLocalVue();
    Vue.use(Buefy);
    Vue.use(Vuex);
    Vue.use(VueRouter);
    localVue = Vue;
  });

  test('render component', () => {
    const wrapper = shallowMount(ShortGenerator, {
      localVue,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('Exist itineraryComponent if tagsAdded is true', () => {
    const wrapper = shallowMount(ShortGenerator, {
      localVue,
      data() {
        return {
          tagsAdded: true,
        };
      },
    });

    expect(wrapper.find('.itineraryComponent').exists()).toBeTruthy();
  });

  test('Not exist itineraryComponent if tagsAdded is false', () => {
    const store = new Vuex.Store(cloneDeep(storeConfig));
    const wrapper = shallowMount(ShortGenerator, {
      localVue,
      store,
    });
    expect(wrapper.html()).toBeFalsy();
  });

  test('Checks ideaItem props', () => {
    const wrapper = shallowMount(ShortGenerator, {
      localVue,
      data() {
        return {
          ideaStructure: [mock],
          tagsAdded: true,
        };
      },
    });
    const attr = wrapper.find('idea-item-stub').attributes();

    expect(attr.description).toBe(mock.description);
    expect(attr.end).toBe(mock.timeEnd.toString());
    expect(attr.start).toBe(mock.timeStart.toString());
    expect(attr.name).toBe(mock.name);
  });
});
