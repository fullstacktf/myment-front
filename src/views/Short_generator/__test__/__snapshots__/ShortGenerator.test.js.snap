// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortGenerator Not exist itineraryComponent if tagsAdded is false 1`] = `
"<div class=\\"columns bigParent\\"><div class=\\"column void1\\"><router-link-stub to=\\"/\\" tag=\\"a\\" event=\\"click\\"><img src=\\"../../../public/assets/myment-logo.png\\" alt=\\"MYMENT\\" class=\\"is-128x128\\"></router-link-stub></div> <div class=\\"column centerzone is-10\\"><b-field-stub addons=\\"true\\" class=\\"topField\\"><b-taginput-stub usehtml5validation=\\"true\\" value=\\"\\" data=\\"\\" type=\\"is-dark\\" hascounter=\\"true\\" field=\\"value\\" closable=\\"true\\" confirmkeycodes=\\"13,188\\" removeonkeys=\\"8\\" onpasteseparators=\\",\\" beforeadding=\\"function _default() {
        return true;
      }\\" placeholder=\\"Enter the location in 2 tags \\"></b-taginput-stub> <b-radio-button-stub type=\\"is-primary\\">search location</b-radio-button-stub></b-field-stub> <div class=\\"tabWrapper\\"><b-tabs-stub expanded=\\"true\\" type=\\"is-toggle is-danger \\" animated=\\"true\\"><b-tab-item-stub label=\\"short\\" icon=\\"video\\" visible=\\"true\\"><div class=\\"masterContent\\"><div class=\\"itineraryComponent\\"><div class=\\"ideaTableShort\\"><idea-item-stub name=\\"hotel\\" description=\\"lorem ipsum ...\\" start=\\"0\\" end=\\"0\\" location=\\"https://www.google.es\\"></idea-item-stub><idea-item-stub name=\\"hotel\\" description=\\"lorem ipsum ...\\" start=\\"0\\" end=\\"0\\" location=\\"#\\"></idea-item-stub><idea-item-stub name=\\"hotel\\" description=\\"lorem ipsum ...\\" start=\\"0\\" end=\\"0\\" location=\\"#\\"></idea-item-stub><idea-item-stub name=\\"hotel\\" description=\\"lorem ipsum ...\\" start=\\"0\\" end=\\"0\\" location=\\"#\\"></idea-item-stub><idea-item-stub name=\\"hotel\\" description=\\"lorem ipsum ...\\" start=\\"0\\" end=\\"0\\" location=\\"#\\"></idea-item-stub><idea-item-stub name=\\"hotel\\" description=\\"lorem ipsum ...\\" start=\\"0\\" end=\\"0\\" location=\\"#\\"></idea-item-stub><idea-item-stub name=\\"hotel\\" description=\\"lorem ipsum ...\\" start=\\"0\\" end=\\"0\\" location=\\"#\\"></idea-item-stub><idea-item-stub name=\\"hotel\\" description=\\"lorem ipsum ...\\" start=\\"0\\" end=\\"0\\" location=\\"#\\"></idea-item-stub><idea-item-stub name=\\"hotel\\" description=\\"lorem ipsum ...\\" start=\\"0\\" end=\\"0\\" location=\\"#\\"></idea-item-stub><idea-item-stub name=\\"hotel\\" description=\\"lorem ipsum ...\\" start=\\"0\\" end=\\"0\\" location=\\"#\\"></idea-item-stub></div></div></div></b-tab-item-stub> <b-tab-item-stub label=\\"custom\\" icon=\\"library-music\\" visible=\\"true\\"><div class=\\"ideaTableCustom\\"><h1>custom itinerary not available</h1></div></b-tab-item-stub> <b-tab-item-stub label=\\"free\\" icon=\\"video\\" visible=\\"true\\"><div class=\\"ideaTableFree\\"><h1>free itinerary not available</h1></div></b-tab-item-stub></b-tabs-stub></div></div> <div class=\\"column void2\\"><router-link-stub to=\\"/home\\" tag=\\"a\\" event=\\"click\\"><!----></router-link-stub></div></div>"
`;

exports[`ShortGenerator render component 1`] = `
"<div class=\\"columns bigParent\\"><div class=\\"column void1\\"><router-link-stub to=\\"/\\" tag=\\"a\\" event=\\"click\\"><img src=\\"../../../public/assets/myment-logo.png\\" alt=\\"MYMENT\\" class=\\"is-128x128\\"></router-link-stub></div> <div class=\\"column centerzone is-10\\"><b-field-stub addons=\\"true\\" class=\\"topField\\"><b-taginput-stub usehtml5validation=\\"true\\" value=\\"\\" data=\\"\\" type=\\"is-dark\\" hascounter=\\"true\\" field=\\"value\\" closable=\\"true\\" confirmkeycodes=\\"13,188\\" removeonkeys=\\"8\\" onpasteseparators=\\",\\" beforeadding=\\"function _default() {
        return true;
      }\\" placeholder=\\"Enter the location in 2 tags \\"></b-taginput-stub> <b-radio-button-stub type=\\"is-primary\\">search location</b-radio-button-stub></b-field-stub> <div class=\\"tabWrapper\\"><b-tabs-stub expanded=\\"true\\" type=\\"is-toggle is-danger \\" animated=\\"true\\"><b-tab-item-stub label=\\"short\\" icon=\\"video\\" visible=\\"true\\"><div class=\\"masterContent\\"><div class=\\"itineraryComponent\\"><div class=\\"ideaTableShort\\"><idea-item-stub name=\\"hotel\\" description=\\"lorem ipsum ...\\" start=\\"0\\" end=\\"0\\" location=\\"https://www.google.es\\"></idea-item-stub><idea-item-stub name=\\"hotel\\" description=\\"lorem ipsum ...\\" start=\\"0\\" end=\\"0\\" location=\\"#\\"></idea-item-stub><idea-item-stub name=\\"hotel\\" description=\\"lorem ipsum ...\\" start=\\"0\\" end=\\"0\\" location=\\"#\\"></idea-item-stub><idea-item-stub name=\\"hotel\\" description=\\"lorem ipsum ...\\" start=\\"0\\" end=\\"0\\" location=\\"#\\"></idea-item-stub><idea-item-stub name=\\"hotel\\" description=\\"lorem ipsum ...\\" start=\\"0\\" end=\\"0\\" location=\\"#\\"></idea-item-stub><idea-item-stub name=\\"hotel\\" description=\\"lorem ipsum ...\\" start=\\"0\\" end=\\"0\\" location=\\"#\\"></idea-item-stub><idea-item-stub name=\\"hotel\\" description=\\"lorem ipsum ...\\" start=\\"0\\" end=\\"0\\" location=\\"#\\"></idea-item-stub><idea-item-stub name=\\"hotel\\" description=\\"lorem ipsum ...\\" start=\\"0\\" end=\\"0\\" location=\\"#\\"></idea-item-stub><idea-item-stub name=\\"hotel\\" description=\\"lorem ipsum ...\\" start=\\"0\\" end=\\"0\\" location=\\"#\\"></idea-item-stub><idea-item-stub name=\\"hotel\\" description=\\"lorem ipsum ...\\" start=\\"0\\" end=\\"0\\" location=\\"#\\"></idea-item-stub></div></div></div></b-tab-item-stub> <b-tab-item-stub label=\\"custom\\" icon=\\"library-music\\" visible=\\"true\\"><div class=\\"ideaTableCustom\\"><h1>custom itinerary not available</h1></div></b-tab-item-stub> <b-tab-item-stub label=\\"free\\" icon=\\"video\\" visible=\\"true\\"><div class=\\"ideaTableFree\\"><h1>free itinerary not available</h1></div></b-tab-item-stub></b-tabs-stub></div></div> <div class=\\"column void2\\"><router-link-stub to=\\"/home\\" tag=\\"a\\" event=\\"click\\"><!----></router-link-stub></div></div>"
`;
